rules/use-external-link.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+const astUtils = require('jsx-ast-utils')
+const getElementType = require('eslint-plugin-jsx-a11y/lib/util/getElementType')
+
+module.exports =  {
+    meta: {
+        docs: {
+            description: 'Enforce all anchors with target="_blank" to use ExternalLink component',
+        },
+    },
+
+create: (context) => {
+    const elementType = getElementType(context);
+        return {
+            JSXOpeningElement: (node) => {
+                const { attributes } = node;
+                const typeCheck = 'a';
+                const nodeType = elementType(node);
+                // Only check anchor elements
+                if (!nodeType || typeCheck !== nodeType) {
+                    return;
+                }
+
+                const propsToValidate = ['target'];
+                const values = propsToValidate.map((prop) => astUtils.getPropValue(astUtils.getProp(node.attributes, prop)));
+                // Checks if the target attribute is set to _blank (ie, is an external link)
+                const hasAnyTarget = values.some((value) => value != null && value === '_blank');
+                // Need to check for spread operator as props can be spread onto the element
+                // leading to an incorrect validation error.
+                const hasSpreadOperator = attributes.some((prop) => prop.type === 'JSXSpreadAttribute');
```

Comment:
What is the intent for spread operator? I'm assuming the intent is that if there is a spread operator that it might contain a `target=_blank` and flag it as should use ExternalLink, even though we might get some false positives, which is fine by me.

If that is the intent, I think the check down below should be `if (!hasBlankTarget && !hasSpreadOperator) {`

Reply:
The intent was actually the opposite, to avoid false positives. I'm 0/5 on which way it should be. @hmhealey opinions?

Reply:
ahh, ok. In that case we might want an adjustment so that if it has a blank target **and** a spread operator, it still triggers the rule.

Reply:
@neallred I've adjusted to only allow the spread operator if there is no explicit blank target. Lmk what you think?
```
                if (!hasBlankTarget && hasSpreadOperator) {
                    return;
                }
```

Reply:
Sorry that I missed the conversation on that. I think that's fine if we don't use the rule when there's a spread operator there. While we might be forgetting to use ExternalLink in some cases, I think that's probably fine. It's not security-critical or anything at the moment